File: sample-videochat-webrtc/src/main/java/com/quickblox/sample/groupchatwebrtc/utils/WebRtcSessionManager.java
package com.quickblox.sample.groupchatwebrtc.utils;

import android.util.Log;

import com.quickblox.videochat.webrtc.QBRTCClient;
import com.quickblox.videochat.webrtc.QBRTCSession;
import com.quickblox.videochat.webrtc.callbacks.QBRTCClientSessionCallbacksImpl;

/**
 * Created by tereha on 16.05.16.
 */
public class WebRtcSessionManager extends QBRTCClientSessionCallbacksImpl {
    private static final String TAG = WebRtcSessionManager.class.getSimpleName();

    public static WebRtcSessionManager instance;

    private static QBRTCSession currentSession;

    public static WebRtcSessionManager getInstance(){
        if (instance == null){
            instance = new WebRtcSessionManager();
        }

        return instance;
    }

    public QBRTCSession getCurrentSession() {
        return currentSession;
    }

    public void setCurrentSession(QBRTCSession qbCurrentSession) {
        currentSession = qbCurrentSession;
    }

    @Override
    public void onReceiveNewSession(QBRTCSession session) {
        super.onReceiveNewSession(session);
        Log.d(TAG, "onReceiveNewSession to WebRtcSessionManager");

        if (currentSession == null){
            setCurrentSession(session);
        }
    }

    @Override
    public void onSessionClosed(QBRTCSession session) {
        super.onSessionClosed(session);
        Log.d(TAG, "onSessionClosed WebRtcSessionManager");

        if (session.equals(getCurrentSession())){
            setCurrentSession(null);
        }
    }
}

File: sample-videochat-kotlin/app/src/main/java/com/quickblox/sample/videochat/kotlin/utils/WebRtcSessionManager.kt
package com.quickblox.sample.videochat.kotlin.utils

import android.util.Log
import com.quickblox.sample.videochat.kotlin.App
import com.quickblox.sample.videochat.kotlin.activities.CallActivity
import com.quickblox.videochat.webrtc.QBRTCSession
import com.quickblox.videochat.webrtc.callbacks.QBRTCClientSessionCallbacksImpl

object WebRtcSessionManager : QBRTCClientSessionCallbacksImpl() {
    private val TAG = WebRtcSessionManager::class.java.simpleName

    private var currentSession: QBRTCSession? = null

    fun getCurrentSession(): QBRTCSession? {
        return currentSession
    }

    fun setCurrentSession(qbCurrentSession: QBRTCSession?) {
        currentSession = qbCurrentSession
    }

    override fun onReceiveNewSession(session: QBRTCSession) {
        Log.d(TAG, "onReceiveNewSession to WebRtcSessionManager")

        if (currentSession == null) {
            setCurrentSession(session)
            CallActivity.start(App.getInstance(), true)
        }
    }

    override fun onSessionClosed(session: QBRTCSession?) {
        Log.d(TAG, "onSessionClosed WebRtcSessionManager")

        if (session == getCurrentSession()) {
            setCurrentSession(null)
        }
    }
}